### DISCUSSION TITLE
What's the humidity in area? - multiple sensors. How to respond?

### DISCUSSION THREAD

**Discussion by tetele** [OHF-Voice/intents#1452] [_Polls]
This is an example, but the same question can be applied to other classes of numeric entities (e.g. temperature, precipitation level, anything really).

There are multiple references here:
- https://github.com/home-assistant/intents/discussions/1204
- https://github.com/home-assistant/intents/issues/1186#issuecomment-1642298826
- others I can't find at the moment

Please comment with your reasoning for picking an option.

**Comment by spuljko** [comment]
The average value for sure isn't good solution in my opinion. 
Temperature for example can be weather temperature, area temperature, or some device temperature (fridge, oven, water heater, aquarium, 3d printer). 
Let's say that I have 3 temperature sensors in kitchen, one that measures area temperature, one for fridge and one for oven. The average temperature information is not very useful. 
If we could somehow make distinct from area and device temperature that would be great, maybe that is something to be considered as HA improvement.

**Reply by tetele** [reply]
There is no such thing as an "area temperature". Even if you have an air temperature sensor in the kitchen, that in itself is a device which exposes a `sensor` entity with `device_class=temperature`. You might have a helper or a template sensor that's not tied to any device, but that doesn't make it an "area" sensor.

Then again, I was also thinking about sentences for `what is the <device name> <class>?`, e.g. "what is the fridge temperature?", "what is the living room air conditioner power" etc. Although it doesn't solve the issues you depicted, it could narrow down on what temperature/power/anything you want to hear about. What do you think about this approach?

Also, are you saying that none of the proposed variants make sense?

**Reply by spuljko** [reply]
@tetele I've wrote that temperature **can be**  (IRL) area or device temperature. I'm also aware that temperature sensor can be entity of other device (domain) like climate (AC, TVR etc) - and in that case it is "area temperature".

> Then again, I was also thinking about sentences for "what is the ?", e.g. "what is the fridge temperature?", "what is the living room air conditioner power" etc. Although it doesn't solve the issues you depicted, it could narrow down on what temperature/power/anything you want to hear about. What do you think about this approach?

I can't see any problem with that, isn't it working that way already? But, now that you've mention that, it would be nice if we could ask something like "What is the average|minimum|maximim room|fridge|plant [sensor] temperature|humidity?", "How much power are my air conditioners using?" What do you think?

> Also, are you saying that none of the proposed variants make sense?

No, I am saying that average response doesn't make sense.

**Reply by tetele** [reply]
Oops, seems that not all of my message came across due to formatting. I've edited the initial comment to fix that. The proposal was `what is the <device name> <device_class>?`. We don't have device querying at the moment, only entity (by name or alias) and area (also by name or alias).

> it would be nice if we could ask something like "What is the average|minimum|maximim room|fridge|plant [sensor] temperature|humidity?", "How much power are my air conditioners using?"

The explicit mathematical operators (e.g. average, minimum etc.) are a great idea IMHO! The rest of it is basically what I was proposing but got cut off because of the `<` and `>` brackets.

So, to reiterate, if a device (let's call it "living room air conditioner") exposes multiple sensors with various `device_class`es, it would be nice to query by device name and extract only those sensors with matching `device_class` (e.g. `power` or `temperature`). Additionally, it would be nice indeed to be able to perform comparisons/averages on a set of matched states and provide whatever the user explicitly asked for (i.e. average, min/max, range etc.).

The second part is already doable. The first part (querying by device name) requires some additions in HA and I'd like to get @synesthesiam 's and @balloob 's point of view on this.
